#158487972/Users should be able to chat

phemonick opened the pull request:
#### What does this PR do?
Create user chat feature
#### Description of Task to be completed?
Users should be able to chat with the group while they are in a game
#### How should this be manually tested?
N/A
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#158487972
#### Screenshots (if appropriate)
#### Questions:
<img width="374" alt="screen shot 2018-07-16 at 3 08 46 pm" src="https://user-images.githubusercontent.com/23261181/42767475-0517dc9c-8915-11e8-84ef-f63fefcafb38.png">

otseobande commented on client/js/app.js:
```diff
@@ -47,6 +47,17 @@ angular.module('mean',
+  //     messagingSenderId: '332516522713'
+  //   };
+  //   firebase.initializeApp(initial);
+  // })
```
If we are not using this code wouldn't it be better we remove it?

phemonick commented on client/js/controllers/chat.js:
```diff
@@ -0,0 +1,84 @@
+  .controller('ChatController', [
+    '$scope', '$firebaseArray', 'game',
+    ($scope, $firebaseArray, game) => {
+      const ref = new Firebase(`https://gondor-cfh.firebaseio.com/${
```
Firebase is injected in the global scope

otseobande commented on client/views/app.html:
```diff
@@ -80,6 +80,9 @@ <h5>Please choose a region below so we can know how horrible you are</h5>
     <answers></answers>
+    <div>
+      <chat></chat>
+    </div>
```
Does the `div` around the `<chat>` directive play any role? I feel it can be moved to the template file of the directive.

otseobande commented on server/src/views/includes/foot.jade:
```diff
@@ -39,6 +44,10 @@ script(type='text/javascript', src='/js/controllers/game.js')
 script(type='text/javascript', src='/js/controllers/profile.js')
 script(type='text/javascript', src='/js/init.js')
+script(type='text/javascript', src='/js/firbase.config.js')
+script(type='text/javascript', src='/js/controllers/chat.js')
```
To maintain readability I feel the chat controller declaration can be moved up the script declaration chain so it can be alongside other controllers.

otseobande reviewed (commented):
Nice implementation of the chat feature. Please go through the comments I made on some files. Also, take note of the linting comments left by Hound.

otseobande commented:
![chat feedback](https://user-images.githubusercontent.com/17123317/42770484-b9493182-891c-11e8-8df1-b038eb2dde2e.png)

The circled areas are the only places that trigger the close and open of the chat window. I feel the whole header area should trigger this.

otseobande commented:
It seems like I can seem like I can view messages from other games in a new game. Is this how the feature should be?

ozimos commented on client/views/chat.html:
```diff
@@ -0,0 +1,44 @@
+                <input ng-disabled="disable" class="text-message" ng-model="message.text" ng-change="messages.$save(message)"/>
+              </div>
+              <div ng-if="name == message.username" class="chat-s">
+                <!-- <i ng-click="edit()" class="material-icons">edit</i> -->
```
If not needed (line 31) you may want to remove. Also, some of your class names are not descriptive enough(Line 23 and 24)

VictoriaOffoma replied:
Please attend to Moses' comment.

phemonick replied:
thanks for the feedback, have made them more descriptive.

ozimos reviewed (commented):
I highlighted some issues with commenting and descriptive class names. Please check the comments.

VictoriaOffoma commented on server/src/socket/socket.js:
```diff
@@ -182,8 +183,8 @@ export default (io) => {
     if (gamesNeedingPlayers.length <= 0) {
-      gameID += 1;
-      const gameIDStr = gameID.toString();
+      // gameID += 1;
```
Please remove this commented code at line 186.

phemonick replied:
ok

Commit:
feat(chat): users group chat
- Set up firebase for data storage
- Create controller to handle firebase
- create controller function to handle chat histories
- create mockup for chat
- implement mockup on angularjs
- create delete chat feature
